internal/dataplane/sendconfig/sendconfig.go

```diff
@@ -49,8 +51,22 @@ func PerformUpdate(ctx context.Context,
 			if !hasSHAUpdateAlreadyBeenReported(newSHA) {
 				log.Debugf("sha %s has been reported", hex.EncodeToString(newSHA))
 			}
-			log.Debug("no configuration change, skipping sync to kong")
-			return oldSHA, nil
+			// we assume ready as not all Kong versions provide their configuration hash, and their readiness state
+			// is always unknown
+			ready := true
+			status, err := kongConfig.Client.Status(ctx)
+			if err != nil {
+				log.Errorf("checking config status failed: %w", err)
+				log.Debug("configuration state unknown, skipping sync to kong")
```

Comment:
Are there any situations you can think of where the status would return an error, but it would still be otherwise possible to push `/config` to the admin API? :thinking:

Reply:
In practice, probably not really. Status shouldn't normally fail unless something something is quite wrong (to the point where Kong APIs aren't able to respond to much of anything) or Kong is unreachable over the network, and it failing implies impending doom for the Pod because it's also the liveness check.

Still, seems prudent to log it/

Reply:
:+1: